### PR TITLE
Raise exceptions if SSH command fail

### PR DESCRIPTION
**Pull request by hsong-rh**
https://github.com/ManageIQ/manageiq-providers-amazon/issues/404

### CODE REVIEW

**Comment by hsong-rh** [comment]
@agrare @bronaghs @jerryk55 @roliveri Please review.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb [hunk 1]
```diff
@@ -59,7 +59,6 @@ def startup_agent
     agent_ids.empty? ? deploy_agent : activate_agents
   rescue => err
     _log.error("No agent is set up to process requests: #{err.message}")
-    _log.error(err.backtrace.join("\n"))
```
Comment (roliveri) [inline app/models/manageiq/providers/amazon/agent_coordinator.rb]: Should we remove this backtrace, or just change it to `_log.error`?

Comment (hsong-rh) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: Most time this backtrace is duplicated with https://github.com/hsong-rh/manageiq-providers-amazon/blob/530f58b7bf0df538fe99859b0dd71cf6bf973efa/app/models/manageiq/providers/amazon/agent_coordinator.rb#L208

File: app/models/manageiq/providers/amazon/agent_coordinator.rb [hunk 2]
```diff
@@ -204,6 +204,11 @@ def deploy_agent
     _log.info("Docker #{docker_image} is loaded. Start to heartbeat.")
 
     instance.id
+  rescue => err
+    _log.error(err.backtrace.join("\n"))
```
Comment (roliveri) [inline app/models/manageiq/providers/amazon/agent_coordinator.rb]: It seems we're logging the backtrace for the error, but we're not logging the actual error itself.
True, we do re-raise a new error with the original message, but that will be logged after the backtrace, possibly leading to confusion.

Will the re-raised error be rescued and logged as well? If so, maybe we should just log the backtrace there.
If you want to log the context at this point, I would log the `err.message` before the backtrace.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb (hunk 1 above)
Comment (roliveri) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: OK.

File: app/models/manageiq/providers/amazon/agent_coordinator.rb (hunk 2 above)
Comment (hsong-rh) [reply app/models/manageiq/providers/amazon/agent_coordinator.rb]: Changed as you suggested.

**Pull request merged**